### ISSUE TITLE
Magic getter makes psalm not throw an error when you access an undefined property

### ISSUE THREAD

**Issue by nickyr** [vimeo/psalm#499]
This should throw an error on `$this->foo` since foo is not defined. If you delete the magic getter, it does throw an error.

```php
<?php
class A {
  public function __get(string $name): void {
  }
  
  public function badGet(): void {
    echo (string) $this->foo;
  }
}
```

**Comment by weirdan** [comment]
> since foo is not defined
Could you elaborate? To me, having `__get()` defined on a class causes any properties to be defined. Or are you referring to this specific form of `__get()` (`void` return type?)?

**Comment by nickyr** [comment]
Sorry the void return type was a typo. Since foo is being accessed directly _from within the class_, the magic getter will not be used in the badGet function. So foo actually needs to be defined on the object.

**Comment by weirdan** [comment]
> the magic getter will not be used in the badGet function

It does get invoked: https://3v4l.org/9XGQJ

**Comment by muglug** [comment]
Turns out Psalm is correct. Which surprised me!

**Comment by nickyr** [comment]
LOL so PHP is like there's no instance variable, guess I'll try the magic getter? Classic. Thanks for catching, @weirdan!
